cmd/pbstatuser/main.go

```diff
@@ -147,30 +156,34 @@ func checkSourceAvailabilityGCP(ctx context.Context) {
 	defer processingWG.Done()
 
 	for s := range chGcp {
-		sr := kafka.SourceResult{
-			SourceID:     s.SourceID,
-			Identity:     s.Identity,
-			ResourceType: "Source",
-		}
-		gcpClient, err := clients.GetGCPClient(ctx, &s.Authentication)
-		if err != nil {
-			sr.Status = kafka.StatusUnavailable
-			sr.Err = err
-			logger.Warn().Msgf("Could not get gcp client %s", err)
-			chSend <- sr
-			continue
-		}
-		_, err = gcpClient.ListAllRegions(ctx)
-		if err != nil {
-			sr.Status = kafka.StatusUnavailable
-			sr.Err = err
-			logger.Warn().Msgf("Could not list gcp regions %s", err)
-			chSend <- sr
-		} else {
-			sr.Status = kafka.StatusAvaliable
-			chSend <- sr
-		}
-		metrics.IncTotalAvailablilityCheckReqs(models.ProviderTypeGCP, "statuser", sr.Status, err)
+		metrics.ObserveAvailablilityCheckReqsDuration(models.ProviderTypeGCP, func() error {
+			var err error
+			sr := kafka.SourceResult{
+				SourceID:     s.SourceID,
+				Identity:     s.Identity,
+				ResourceType: "Source",
+			}
+			gcpClient, err := clients.GetGCPClient(ctx, &s.Authentication)
+			if err != nil {
+				sr.Status = kafka.StatusUnavailable
+				sr.Err = err
+				logger.Warn().Msgf("Could not get gcp client %s", err)
+				chSend <- sr
+			}
+			_, err = gcpClient.ListAllRegions(ctx)
+			if err != nil {
+				sr.Status = kafka.StatusUnavailable
+				sr.Err = err
+				logger.Warn().Msgf("Could not list gcp regions %s", err)
+				chSend <- sr
+			} else {
+				sr.Status = kafka.StatusAvaliable
+				chSend <- sr
+			}
+			metrics.IncTotalAvailabilityCheckReqs(models.ProviderTypeGCP, "statuser", sr.Status, err)
+
+			return fmt.Errorf("An error has occured %w:", err)
```

Comment:
```suggestion
			return fmt.Errorf("an error has occurred %w:", err)
```

(The linter should complain)